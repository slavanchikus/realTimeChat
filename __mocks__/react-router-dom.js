import React from 'react';

const rrd = require('react-router-dom');
rrd.HashRouter = ({ children }) => <div>{children}</div>;

module.exports = rrd;
