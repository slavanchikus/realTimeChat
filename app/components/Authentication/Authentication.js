import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './Authentication.module.styl';

export default class Authentication extends Component {
  static propTypes = {
    onUserRequest: PropTypes.func.isRequired,
    onUserCreate: PropTypes.func.isRequired,
  };

  state = {
    username: '',
    password: '',
    isRegistration: false
  };

  handleClick = () => {
    const { username, password, isRegistration } = this.state;
    if (isRegistration) {
      this.props.onUserCreate(username, password);
    } else {
      this.props.onUserRequest(username, password);
    }
  };

  putUsername = (e) => {
    this.setState({ username: e.target.value });
  };

  putPassword = (e) => {
    this.setState({ password: e.target.value });
  };

  render() {
    const { isRegistration } = this.state;
    return (
      <div className={styles.container}>
        <h2>
          {isRegistration ? 'Регистрация' : 'Авторизация'}
        </h2>
        <input type="text" placeholder="Юзернейм" onChange={this.putUsername} />
        <input type="text" placeholder="Пароль" onChange={this.putPassword} />
        <div className={styles.button} onClick={this.handleClick}>
          {isRegistration ? 'Регистрировать' : 'Войти'}
        </div>
        <div className={styles.choose} onClick={() => this.setState({ isRegistration: !this.state.isRegistration })}>
          {isRegistration ? 'Авторизоваться' : 'Зарегистрироваться'}
        </div>
      </div>
    );
  }
}
