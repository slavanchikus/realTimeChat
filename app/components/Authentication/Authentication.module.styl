.container
  height: 200px
  width: 300px
  padding: 35px 0
  background: #181818
  display: flex
  align-items: center
  flex-direction: column
  justify-content: space-around
  font-size: 13px
  input
    height: 22px
    padding: 4px
    border: 0
    outline: 0
    border-radius: 2px
  h2
    margin: 0

.button
  width: 177px
  height: 30px
  text-align: center
  line-height: 30px
  background: #ffdb4d
  border-color: rgba(191,153,0,.8)
  color: black
  cursor: pointer
  border-radius: 2px
  transition-duration: .2s;
  &:hover
    background: #ECC948

.choose
  margin-top: 10px
  cursor: pointer
  &:hover
    color: #FF6557
