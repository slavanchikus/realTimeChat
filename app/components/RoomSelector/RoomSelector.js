import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styles from './RoomSelector.module.styl';

export default class RoomSelector extends PureComponent {
  static propTypes = {
    username: PropTypes.string.isRequired,
    allRooms: PropTypes.array.isRequired,
    onGetMessages: PropTypes.func.isRequired
  };

  handleClick = (id) => {
    console.log(id);
    this.props.onGetMessages(0, this.props.username, id);
  };

  render() {
    const { allRooms } = this.props;
    return (
      <div className={styles.container}>
        {allRooms.map(item =>
          <div
            key={item.roomName}
            className={styles.room}
            onClick={() => this.handleClick(item._id)}
          >
            <h2>{item.roomName}</h2>
            <div>{item.description}</div>
          </div>
        )}
      </div>
    );
  }
}
