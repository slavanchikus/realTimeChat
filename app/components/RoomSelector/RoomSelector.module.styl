.container
  display: flex
  flex-direction: column
  justify-content: flex-start
  padding: 15px
  width: 600px
  height: 100%
  box-sizing: border-box
  background: #181818
  border: 1px solid
  border-color: hsla(0,0%,100%,0.15)
  border-radius: 10px
  font-size: 13px
  h2
   margin: 0

.room
  padding: 10px
  border-bottom: 1px solid
  cursor: pointer
  user-select: none
  &:hover
    border-color: #ffdb4d

@media (max-width: 660px)
  .container
    width: 100%
