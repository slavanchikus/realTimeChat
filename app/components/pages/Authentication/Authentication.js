import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ErrorContainer from '../../shared/ErrorContainer/ErrorContainer';

import styles from './Authentication.module.styl';

export default class Authentication extends PureComponent {
  static propTypes = {
    errors: PropTypes.object.isRequired,
    onUserRequest: PropTypes.func.isRequired,
    onUserCreate: PropTypes.func.isRequired,
  };

  state = {
    username: '',
    password: '',
    isRegistration: false
  };

  handleClick = () => {
    const { username, password, isRegistration } = this.state;
    const usernameLen = !/^\s+$/.test(username) ? username.length : 0;
    const passwordLen = !/^\s+$/.test(password) ? password.length : 0;
    if (usernameLen > 1 && passwordLen > 1) {
      if (isRegistration) {
        this.props.onUserCreate(username, password);
      } else {
        this.props.onUserRequest(username, password);
      }
    }
  };

  putUsername = (e) => {
    this.setState({ username: e.target.value });
  };

  putPassword = (e) => {
    this.setState({ password: e.target.value });
  };

  render() {
    const { errors } = this.props;
    const { isRegistration, username, password } = this.state;
    return (
      <div className={styles.container}>
        <h2>
          {isRegistration ? 'Регистрация' : 'Авторизация'}
        </h2>
        <input type="text" placeholder="Юзернейм" onChange={this.putUsername} value={username} maxLength={15} />
        <input type="password" placeholder="Пароль" onChange={this.putPassword} value={password} maxLength={10} />
        <div className={styles.button} onClick={this.handleClick}>
          {isRegistration ? 'Регистрировать' : 'Войти'}
        </div>
        <div className={styles.choose} onClick={() => this.setState({ isRegistration: !this.state.isRegistration, username: '', password: '' })}>
          {isRegistration ? 'Авторизоваться' : 'Зарегистрироваться'}
        </div>
        {errors.userError &&
          <ErrorContainer
            error={errors.userError}
          />}
      </div>
    );
  }
}
