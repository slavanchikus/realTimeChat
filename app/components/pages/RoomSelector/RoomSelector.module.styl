.container
  display: flex
  flex-direction: column
  justify-content: flex-start
  width: 600px
  height: 100%
  position: relative
  box-sizing: border-box
  background: #181818
  border: 1px solid
  border-color: hsla(0,0%,100%,0.15)
  border-radius: 10px
  font-size: 13px
  h2, h3
   margin: 0

.header
  display: flex
  flex-direction: row
  justify-content: space-between
  align-items: center
  padding: 15px
  border-bottom: 1px solid #ffdb4d

.plus
  font-size: 30px
  cursor: pointer
  &:hover
    color: #ffdb4d

.rooms
  padding: 15px
  div
    &:first-child
      margin-top: 0


@media (max-width: 660px)
  .container
    width: 100%
