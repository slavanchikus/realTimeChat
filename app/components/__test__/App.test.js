import React from 'react';
import { shallow, configure } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import App from '../App';
import Authentication from '../pages/Authentication/Authentication';
import ChatWrapper from '../pages/ChatWrapper/ChatWrapper';
import RoomSelector from '../pages/RoomSelector/RoomSelector';

configure({ adapter: new Adapter() });

describe('App', () => {
  const store = createMockStore({
    user: {
      userId: '5a66ea7ba831d74870af4885',
      username: 'test',
      lastViewedMessage: 1519307926006
    },
    rooms: {
      allRooms: {
        _id: '5abba5f56243f5939c7b17b5',
        roomName: '123',
        description: '123',
        participants: [],
        backgroundSrc: '',
        userId: '5a66ea7ba831d74870af4885',
        isPrivate: true
      },
      selectedRoom: {}
    },
    messages: []
  });
  const context = { store };
  let props;
  const appWrapper = newProps => (
    <App {...newProps} />
  );

  describe('when requred props are defined for reply App', () => {
    beforeEach(() => {
      props = { };
    });

    it('renders correctly', () => {
      const wrapper = shallow(appWrapper(props), { context });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('show RoomSelector if user are storaged', () => {
      const wrapper = shallow(
        <MemoryRouter initialEntries={['/#']}>
          {appWrapper(props)}
        </MemoryRouter>
      , { context });
      console.log(wrapper);
    });
  });
});
