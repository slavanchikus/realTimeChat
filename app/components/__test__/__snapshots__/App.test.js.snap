// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when requred props are defined for reply ChatWrapper renders correctly 1`] = `
<MainPage
  changeRoomBackground={[Function]}
  createMessage={[Function]}
  createRoom={[Function]}
  getMessages={[Function]}
  getOneMessage={[Function]}
  messages={Array []}
  openRoom={[Function]}
  removeErrors={[Function]}
  resetRoom={[Function]}
  rooms={
    Object {
      "allRooms": Object {
        "_id": "5abba5f56243f5939c7b17b5",
        "backgroundSrc": "",
        "description": "123",
        "isPrivate": true,
        "participants": Array [],
        "roomName": "123",
        "userId": "5a66ea7ba831d74870af4885",
      },
      "selectedRoom": Object {},
    }
  }
  setRoomBackground={[Function]}
  user={
    Object {
      "lastViewedMessage": 1519307926006,
      "userId": "5a66ea7ba831d74870af4885",
      "username": "test",
    }
  }
  userCreate={[Function]}
  userRequest={[Function]}
/>
`;
